src/SearchField/index.jsx

```diff
@@ -0,0 +1,266 @@
+
+    return (
+      <Button
+        className={[classNames(
+          styles['search-field-search-btn-transition'],
+          {
+            [styles['ml-1']]: !hasClearButton,
+            [styles['text-muted']]: !focused && inputTextHasValue,
+            [styles['border-left']]: !focused && inputTextHasValue,
+            [styles['btn-outline-primary']]: focused && inputTextHasValue,
+          },
+        )]}
+        label={searchButtonLabel}
+        disabled={!this.inputTextHasValue()}
+        inputRef={(input) => { this.searchButton = input; }}
+        onClick={this.handleSubmit}
+      />
+    );
+  }
+
+  getSearchActionButtons() {
+    const buttons = [
+      this.getSearchButton(),
+    ];
+
+    if (this.shouldShowClearButton()) {
+      buttons.unshift(this.getClearButton());
+    }
+
+    return buttons;
+  }
+
+  shouldShowClearButton() {
+    return this.props.hasClearButton && this.inputTextHasValue();
+  }
+
+  inputTextHasValue() {
+    const { value } = this.state;
+    return value && value.trim().length > 0;
+  }
+
+  handleFocus(event) {
+    this.setState({ focused: true });
+    this.props.onFocus(event);
+  }
+
+  handleBlur(event) {
+    this.setState({ focused: false });
+    this.props.onBlur(event);
+  }
+
+  handleChange(value) {
+    this.setState({ value });
+    this.props.onChange(value);
+  }
+
+  handleClear() {
+    this.setState({ value: '' });
+    this.props.onChange('');
+    this.textInput.focus();
+  }
+
+  handleKeyPress(event) {
+    if (event.key === 'Enter' && this.inputTextHasValue()) {
+      this.handleSubmit();
+    }
+  }
+
+  handleSubmit() {
+    const { value } = this.state;
+    this.searchButton.focus();
+    this.props.onSubmit(value);
+  }
+
+  render() {
+    const { focused, value } = this.state;
+    const {
+      screenReaderText,
+      placeholder,
+      roundedCorners,
+      hasClearButton,
+      inputLabel,
+      hideInputLabel,
+    } = this.props;
+    const shouldShowClearButton = this.shouldShowClearButton();
+
+    return (
+      <div
+        className={classNames(
+          styles.border,
+          styles['search-field'],
+          {
+            [styles['border-primary']]: focused,
+            [styles[`search-field-rounded-${roundedCorners}`]]: roundedCorners,
+          },
+        )}
+        onFocus={this.handleFocus}
+        onBlur={this.handleBlur}
+      >
+        <InputText
+          className={[classNames(
+            styles['search-field-input'],
+            {
+              [styles['pl-0']]: !hideInputLabel,
+              [styles['pl-3']]: hideInputLabel,
+              [styles['search-field-input-has-clear-btn']]: hasClearButton && !shouldShowClearButton,
+              [styles['pr-0']]: !hasClearButton || (hasClearButton && shouldShowClearButton),
+            },
+          )]}
+          name="search"
+          label={inputLabel || this.getInputLabel()}
+          ariaLabel={hideInputLabel ? screenReaderText : ''}
```

Comment:
`aria-label`?

Reply:
It turns out using `aria-label` as the prop name does not work as intended.